builder/azure/arm/config.go

```diff
@@ -1241,6 +1246,10 @@ func assertRequiredParametersSet(c *Config, errs *packersdk.MultiError) {
 		}
 	}
 
+	if c.SkipCreateBuildKeyVault && !strings.EqualFold(c.Comm.Type, "winrm") {
```

Comment:
The comm type may not have been normalized at this point, if you check in config.go on line 833 you can see that we don't normalize the comm type's case, so the user may input "WinRM", which equalFold would allow

Reply:
Yeah I was kinda suspecting this, thanks for confirming.
I would maybe suggest we normalise it actively in the `Prepare` step then, or maybe even force specifying it in a set way, not sure automagically accepting all forms of capitalisation for enums is optimal, but we can do a pass on that later, no need to change it today.

Reply:
I think normalizing every variable would be better than not, but I agree it can be fixed in a follow up PR.  Personally I'd be curious if this has caused users any pain so maybe we should search through issues for any instances of this.